weblab/experiments/views.py

```diff
@@ -170,6 +170,93 @@ class ExperimentVersionListView(VisibilityMixin, DetailView):
     template_name = 'experiments/experiment_versions.html'
 
 
+class ExperimentComparisonView(TemplateView):
+    """
+    Compare multiple experiment versions
+    """
+    template_name = 'experiments/experimentversion_compare.html'
+
+    def get_context_data(self, **kwargs):
+        versions = ExperimentVersion.objects.filter(
+            pk__in=map(int, self.kwargs['version_pks'][1:].split('/'))
+        )
+        kwargs.update({
+            'experiment_versions': versions,
+        })
+        return super().get_context_data(**kwargs)
+
+
+class ExperimentComparisonJsonView(View):
+    """
+    Serve up JSON view of multiple experiment versions for comparison
+    """
+    def _file_json(self, version, archive_file):
+        """
+        JSON for a single file in the experiment archive
+
+        :param version: ExperimentVersion object
+        :param archive_file: ArchiveFile object
+        """
+        return {
+            'id': archive_file.name,
+            'author': version.author.full_name,
+            'created': version.created_at,
+            'name': archive_file.name,
+            'filetype': archive_file.fmt,
+            'masterFile': archive_file.is_master,
+            'size': archive_file.size,
+            'url': reverse(
+                'experiments:file_download',
+                args=[version.experiment.id, version.id, urllib.parse.quote(archive_file.name)]
+            )
+        }
+
+    def _version_json(self, version):
+        """
+        JSON for a single experiment version
+
+        :param version: ExperimentVersion object
+        """
+        files = [
+            self._file_json(version, f)
+            for f in version.files
+            if f.name not in ['manifest.xml', 'metadata.rdf']
+        ]
+        return {
+            'id': version.id,
+            'author': version.author.full_name,
+            'status': version.status,
+            'parsedOk': False,
+            'visibility': version.visibility,
+            'created': version.created_at,
+            'name': version.experiment.name,
+            'experimentId': version.experiment.id,
+            'versionId': version.id,
+            'files': files,
+            'numFiles': len(files),
+            'url': reverse(
+                'experiments:version', args=[version.experiment.id, version.id]
+            ),
+            'download_url': reverse(
+                'experiments:archive', args=[version.experiment.id, version.id]
+            ),
+            'modelName': version.experiment.model.name,
+            'protoName': version.experiment.protocol.name,
+        }
+
+    def get(self, request, *args, **kwargs):
+        versions = ExperimentVersion.objects.filter(
+            pk__in=map(int, self.kwargs['version_pks'][1:].split('/'))
```

Comment:
Similarly to above, we should filter this list so only visible versions are returned.